services/controllerhost/consumer.go

```diff
@@ -859,21 +859,21 @@ func refreshOutputHostsForConsGroup(context *Context,
 // shouldConsumeInZone indicated whether we should consume from this zone for a multi_zone consumer group
 // If failover mode is enabled in dynamic config, the active zone will be the one specified in dynamic config
 // Otherwise, use the per cg override if it's specified
-// Last, check the active zone in dynamic config. If specified, use it. Otherwise always return true
+// Last, check the active zone in dynamic config. If specified, use it. Otherwise always return false
 func shouldConsumeInZone(zone string, cgDesc *shared.ConsumerGroupDescription, dConfig ControllerDynamicConfig) bool {
 	if strings.EqualFold(dConfig.FailoverMode, `enabled`) {
 		return strings.EqualFold(zone, dConfig.ActiveZone)
 	}
 
-	if cgDesc.IsSetActiveZone() {
+	if len(cgDesc.GetActiveZone()) > 0 {
 		return strings.EqualFold(zone, cgDesc.GetActiveZone())
 	}
 
 	if len(dConfig.ActiveZone) > 0 {
 		return strings.EqualFold(zone, dConfig.ActiveZone)
 	}
 
-	return true
+	return false
```

Comment:
can you also add a log or metric when both dConfig.ActiveZone and cgDesc.GetActiveZone() are empty ? 
just want to make sure we have mechanisms to verify correctness as we roll this out initially.